connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -296,6 +325,9 @@ private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
     }
 
     private void deliverMessages() {
+        if (messageBatch.isEmpty())
```

Comment:
We should not return here as the connector needs to retry when put() is invoked. So we should call put() even with empty message batch.